pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -104,6 +104,21 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// BIOSConfig contains the configuration that you want to configure BIOS settings in Bare metal server
+type BIOSConfig struct {
+
+	// Select the boot mode of the system. Allowed values are Uefi, LegacyBios
+	// +kubebuilder:validation:Enum=Uefi,LegacyBios
+	BootMode string `json:"bootMode,omitempty"`
```

Comment:
Can we use an enum type internally, too? I'm not sure how well special string types mix with kubebuilder.

Reply:
This field is now handled separately and should be removed here.